xdsl/dialects/builtin.py

```diff
@@ -313,21 +352,39 @@ class FloatAttr(Generic[_FloatAttrTyp], ParametrizedAttribute):
     value: ParameterDef[FloatData]
     type: ParameterDef[_FloatAttrTyp]
 
-    @staticmethod
-    def from_value(
-        value: float, type: _FloatAttrTypContr = Float32Type()
-    ) -> FloatAttr[_FloatAttrTypContr]:
-        return FloatAttr([FloatData.from_float(value), type])
-
-    @staticmethod
+    @overload
+    def __init__(self, data: float | FloatData, type: _FloatAttrTyp) -> None:
+        ...
+
+    @overload
+    def __init__(self, data: float | FloatData, type: int) -> None:
+        ...
+
+    def __init__(self, data: float | FloatData,
+                 type: int | _FloatAttrTyp | AnyFloat) -> None:
```

Comment:
`width` here and above?

Reply:
So I just checked, and we can't have different names between the overloaded functions :(
So I'm not sure if we should have `width` or `type`. I would say to me that `type` would be better, but I don't feel that strong about it, in case you feel strong about it.

Reply:
Nah type is good